### DISCUSSION TITLE
What is the way to setup a PWM counter which triggers IRQ once counter wrapped?

### DISCUSSION THREAD

**Discussion by pchala** [micropython/micropython#16056] [RP2040 / Pico] [answered]
So I would like to after 100 pulses PWM counter trigger IRQ (callback) to handle next step.

I have found https://github.com/phoreglad/pico-MP-modules/blob/main/PWMCounter/README.md but it misses callback functionality.

I can do it with PIO but I would like to save them for other tasks.

**Comment by peterhinch** [comment, marked as answer]
You haven't mentioned the PWM frequency but unless it's very high why not use a counter?
```py
class Foo:
    # Construct with a pin instance configured as an input and your callback
    def __init__(self, pin, callback):
        self.callback = callback
        self.count = 0
        irq = pin.irq(self.cb, Pin.IRQ_RISING, hard=True)

    def cb(self, _):  # Actual pin ISR
        self.count += 1
        self.count %= 100
        if not self.count:
            self.callback()
```

**Reply by pchala** [reply]
Thanks, it is a good idea, I will try to run this in thread.
